test/integration/framework/controlplane_utils.go

```diff
@@ -233,7 +233,10 @@ func startAPIServerOrDie(controlPlaneConfig *controlplane.Config, incomingServer
 	}
 	var lastHealthContent []byte
 	err = wait.PollImmediate(100*time.Millisecond, 30*time.Second, func() (bool, error) {
-		result := privilegedClient.Get().AbsPath("/healthz").Do(context.TODO())
+		ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+		defer cancel()
+
+		result := privilegedClient.Get().AbsPath("/healthz").Do(ctx)
```

Comment:
the change seems OK to me, but normally the `privilegedClient` should return status 200 or an error right away.
did you observe cases where it was blocking for more than 100ms?

Reply:
we have seen it to hang indefinitely here https://github.com/kubernetes/kubernetes/issues/103512. I think the timeout will resolve the indefinite hang issue and make it easier to debug.

Reply:
please link to the issue in the PR description.
also notify the issue discussion participants for a review on this PR.